src/plugins/data/server/index_patterns/fetcher/index_patterns_fetcher.ts

```diff
@@ -48,9 +48,9 @@ interface FieldSubType {
 }
 
 export class IndexPatternsFetcher {
-  private _callDataCluster: LegacyAPICaller | OpenSearchClient;
+  private _callDataCluster: LegacyAPICaller;
 
-  constructor(callDataCluster: LegacyAPICaller | OpenSearchClient) {
```

Comment:
Is changing the valid call signature of this class backwards compatible? Is it possible it would break any plugins attempting to call the `IndexPatternsFetcher` with an`OpenSearchClient`? Or was this change only introduced as part of the initial MD PR?

Reply:
yeah it's mostly a follow-up to the initial MD PR. At that time, since legacy client is not supported by MD, but index_pattern_fetcher is still using legacy client. We had a workaround to enforce index_pattern_fetcher to use new Client when talking to data sources, use legacy client when talk to default cluster. Now with the support of legacy client, we can get rid of the workaround, and keep the type declaration clean, to only use legacy client for any scenario